doc/manual/advanced-topics/post-build-hook.xml

```diff
@@ -122,7 +122,7 @@ post-build-hook = /etc/nix/upload-to-cache.sh
 
   <screen>
 $ nix-build -E '(import &lt;nixpkgs&gt; {}).writeText "example" (builtins.toString builtins.currentTime)'
-these derivations will be built:
+this derivation will be built:
```

Comment:
It seems to me that in this case we should just omit the entire `this derivation will be built` message because it's superfluous. After all the next line already says `building '/nix/store/s4pnfbkalzy5qz57qs6yybna8wylkig6-example.drv'...`.

Reply:
In the case where there is a derivation being built and some being fetched, it distinguishes the build expression from the one being fetched.

Reply:
Also, the derivation may not be displayed promptly if one has to be "waiting for locks or build slots..."